src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -779,7 +779,29 @@ public IEnumerable<Element> GetJoinedElements()
                 .ToList();
         }
 
+        /// <summary>
+        /// Joins two elements geometry if they are intersecting
+        /// </summary>
+        /// <param name="otherElement">Other element to join with</param>
+        /// <returns>The two joined elements</returns>
+        public IEnumerable<Element> JoinGeometry(Element otherElement)
+        {
+            ElementIntersectsElementFilter filter = new ElementIntersectsElementFilter(otherElement.InternalElement);
+            ICollection<Autodesk.Revit.DB.Element> collector = new FilteredElementCollector(Document).WherePasses(filter).ToElements();
+            var joinedElements = new List<Element>() { this, otherElement };
+
+            if (JoinGeometryUtils.AreElementsJoined(Document, this.InternalElement, otherElement.InternalElement))
```

Comment:
this should be the first thing we do, before expensive `filter` or `collector` API calls